packages/clerk-js/src/core/clerk.ts

```diff
@@ -1172,6 +1183,26 @@ export class Clerk implements ClerkInterface {
     return unsubscribe;
   };
 
+  public addStatusListener = (listener: StatusListenerCallback): UnsubscribeCallback => {
+    this.#statusListeners.push(listener);
+    // emit instantly
+    listener(this.status);
+    return () => {
+      this.#statusListeners = this.#statusListeners.filter(l => l !== listener);
+    };
+  };
+
+  public __internal_setStatus = (status: Clerk['status'], opts = { notify: true }): void => {
+    this.#status = status;
+    if (!opts?.notify) {
+      return;
+    }
+
+    for (const listener of this.#statusListeners) {
+      listener(status);
+    }
+  };
```

Comment:
We should encapsulate both the state change and the notification logic so that clients of the class aren’t forced to know about or interact with internal listeners. Instead of exposing an “internal” method that both updates state and conditionally notifies listeners via an options flag, this should be private to the Clerk class to keep the abstraction free of internal implementation details